resolwe_bio/processes/support_processors/bam_conversion.py

```diff
@@ -0,0 +1,152 @@
+        build = StringField(label="Build")
+
+    def run(self, inputs, outputs):
+        """Run the analysis."""
+        path = inputs.alignment.bam.path
+        basename = os.path.basename(path)
+        assert basename.endswith(".bam")
+        name = basename[:-4]
+        bed_file = f"{name}.bed"
+
+        samtools_param = ["-n", path]
+        bedtools_param = ["-bedpe", "-i"]
+
+        (Cmd["samtools"]["sort"][samtools_param] | Cmd["bedtools"]["bamtobed"][bedtools_param] > bed_file)()
+
+        if bed_file not in os.listdir():
+            self.error("BamToBed process failed.")
+
+        outputs.bedpe = bed_file
+        outputs.species = inputs.species
+        outputs.build = inputs.build
+
+
+class ScaleBigWig(Process):
+    """Creates a scaled bigwig file.
+    """
+
+    slug = "scale-bigwig"
+    name = "Normalize bigwig"
+    requirements = {
+        "expression-engine": "jinja",
+        "executor": {"docker": {"image": "resolwebio/rnaseq:4.11.0"}},
+        "resources": {"cores": 1, "memory": 4096},
+    }
+    data_name = "bamCoverage ({{sample_name}})"
+    version = "1.0.0"
+    process_type = "data:coverage:bigwig"
+    category = "Other"
+    entity = {
+        "type": "sample",
+        "input": "alignment",
+    }
+    scheduling_class = SchedulingClass.BATCH
+
+    class Input:
+        """Input fields."""
+
+        alignment = DataField("alignment:bam", label="Alignment BAM file")
+        scale_factor = StringField(label="Scale factor")
+        species = StringField(
+            label="Species",
+            choices=[
+                ("Homo sapiens", "Homo sapiens"),
+                ("Mus musculus", "Mus musculus"),
+                ("Rattus norvegicus", "Rattus norvegicus"),
+                ("Dictyostelium discoideum", "Dictyostelium discoideum"),
+                ("Odocoileus virginianus texanus", "Odocoileus virginianus texanus"),
+                ("Solanum tuberosum", "Solanum tuberosum"),
+            ],
+        )
+        build = StringField(label="Build")
+
+    class Output:
+        """Output fields."""
+
+        bigwig = FileField(label="BEDPE file")
+        species = StringField(label="Species")
+        build = StringField(label="Build")
+
+    def run(self, inputs, outputs):
+        """Run the analysis."""
+        path = inputs.alignment.bam.path
+        basename = os.path.basename(path)
+        assert basename.endswith(".bam")
+        name = basename[:-4]
+        out_file = f"{name}.SInorm.bigwig"
+        out_index = f"{name}.bai"
+
+        bamCoverage_param = [
+            "-b", path,
+            "--scaleFactor", inputs.scale_factor,
+            "-o", out_file,
+            "-p", "30",  # Number of cores
+            "-of", "bigwig",
+        ]
+
+        (Cmd["samtools"]["index"][path][out_index])()
+        (Cmd["bamCoverage"][bamCoverage_param])()
+
+        if out_file not in os.listdir():
+            self.error("bamCoverage process failed.")
```

Comment:
Maybe be more bit explicit - generation of file with bamCoverage filed (or something to that effect).